### PR TITLE
Selva/Ganesh Copying IxPailConsolidator from bluestreak to dfs-datastore

### PR DESCRIPTION
**Pull request by gsriram7**
_Copying the corresponding tests from bluestreak to dfs-datastore
Adding IndixModels, thrift, twitter scalding sbt dependencies_

@sriram-r

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -107,7 +107,11 @@ object Build extends Build {
       "org.apache.hadoop" % "hadoop-client" % "2.0.0-mr1-cdh4.2.1" % "provided",
       "org.apache.hadoop" % "hadoop-core" % "2.0.0-mr1-cdh4.2.1" % "test",
       "com.hadoop.gplcompression" % "hadoop-lzo" % "0.4.15",
-      "org.scalatest" %% "scalatest" % "2.2.0" % "test"
+      "org.scalatest" %% "scalatest" % "2.2.0" % "test",
+      "models" %% "models" % "2.0.175",
```
Comment (ashwanthkumar) [inline project/Build.scala]: This is an open source project, we can't have indix specific stuff added to this. If we need to use indix-models then we might have to make this project private then.

File: dfs-datastores/src/main/scala/com/indix/pail/GenerateFileListForPailCopy.scala [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    files
+  }
+
+  private def siteNameToQualifiedUri(site: String) = {
+    if(site.startsWith("http")) site
+    else "http://%s/".format(site)
+  }
+}
+
+object GenerateFileListForPailCopy {
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/pail/GenerateFileListForPailCopy.scala]: This was used for HDFSRefresh right? Do we still need this?

Comment (gsriram7) [inline dfs-datastores/src/main/scala/com/indix/pail/GenerateFileListForPailCopy.scala]: We just realized that its no longer needed so we removed both the class and its test

File: dfs-datastores/src/main/java/util/DateHelper.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public class DateHelper {
+    public DateHelper() {
+    }
+
+    public static String weekInterval(DateTime var0) {
+        int var1 = var0.getDayOfWeek();
+        String var2 = var0.minusDays(var1 - 1).toString("YYYYMMdd");
+        String var3 = var0.plusDays(7 - var1).toString("YYYYMMdd");
+        return String.format(var2 + "_" + var3, new Object[0]);
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/java/util/DateHelper.java]: Copied from Decompiled version of idea? :P I remember Sriram writing this in Scala.

File: project/assembly.sbt [hunk 4]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.0")
```
Comment (ashwanthkumar) [inline project/assembly.sbt]: Do we need an assembly version of the project?

File: project/Build.scala (hunk 1 above)
Comment (gsriram7) [inline project/Build.scala]: Model dependency was required for DateHelper and Thrift date formatter. We decided to write a local implementation thereby avoid depending indix-models for it. We also feel that having DateHelper and similar utils separated from models and be placed in an exclusive module called *Utils* would make better sense. Any thoughts?

**Comment by gsriram7** [comment]
@sriram-r

File: dfs-datastores/src/main/scala/com/indix/commons/TimerUtil.scala [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package com.indix.commons
+
+object TimerUtil {
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/commons/TimerUtil.scala]: Do we need this class?

File: dfs-datastores/src/main/scala/com/indix/commons/FileHelper.scala [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import java.io._
+import scala.io.Source
+import org.apache.hadoop.fs._
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.io._
+import org.apache.hadoop.fs.FileSystem
+import org.apache.commons.lang3.StringUtils
+
+
+trait FileHelper extends Serializable {
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/commons/FileHelper.scala]: Do we need this trait? I can't find any usages in the PR.

**Comment by sriram-r** [comment]
Why do we need FSUtils, FileHelper and the rest. Looks like, you guys have moved things from bluestreak common to here. We just needed to move PailConsolidate and PailAbsorb programs to here. The only dependency would be DateTimeFormatter and DateHelper. That's it, right? Why so many files.

File: dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import com.backtype.hadoop.pail.Pail
+import com.indix.commons.FSUtils
+import com.twitter.scalding.Args
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.joda.time.DateTime
+import org.slf4j.LoggerFactory
+import util.{DateTimeFormatter, DateHelper}
+
+class PailConsolidate(inputDir: String, subDir: String, pipelineLabel: String) extends FSUtils {
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala]: Are we using any methods from FSUtils in this class?

**Comment by sriram-r** [comment]
There's an `exists` function usage in IxPailConsolidate. I think, we can still make do with having the exists implementation here. It's ok to have such small duplicate code instead of carrying such baggage.

As for moving things into a separate `Util` module, yes, we should. Let's discuss about it on Monday.

File: project/assembly.sbt (hunk 4 above)
Comment (gsriram7) [inline project/assembly.sbt]: Before merging we wanted to build a fat jar and test consolidator in some /tmp.

File: dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala (hunk 7 above)
Comment (gsriram7) [inline dfs-datastores/src/main/scala/com/indix/pail/PailConsolidate.scala]: We use FSUtils' exist method to verify weather a partition exists or not.

**Pull request merged**